pkg/cortex/modules.go

```diff
@@ -517,13 +516,7 @@ func (t *Cortex) initQueryFrontend() (serv services.Service, err error) {
 
 	// Wrap roundtripper into Tripperware.
 	roundTripper = t.QueryFrontendTripperware(roundTripper)
-
-	handler := transport.NewHandler(t.Cfg.Frontend.Handler, roundTripper, util.Logger)
```

Comment:
If we deprecate removing the code as well we introduce a breaking change (which we can't, I think). We should keep the logic as is, deprecated it in the CLI flag description + CHANGELOG and in 2 minor versions we'll remove it. @gouthamve may you confirm it, please?

Reply:
@gotjosh on a related not, does having this flag and the new flag set simultaneously cause an issues? If so you could add a check below to only inject the gzip handler if the query-frontend flag is enabled but not the generic compression flag.

Reply:
> does having this flag and the new flag set simultaneously cause an issues?

We checked when we reviewed the previuos PR (where the new flag has been introduced) and it shouldn't cause any issue.

Reply:
Is this better?